### PR TITLE
Set new --reconfiguration_policy mesos flag.

### PR DESCRIPTION
**Pull request by lava**
## High-level description

This commit sets the newly added reconfiguration policy
for mesos-agent to `additive`, in order to allow agents
to add new domains without draining all tasks.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-1959](https://jira.mesosphere.com/browse/DCOS_OSS-1959)

## Related tickets (optional)

Other tickets related to this change:

  - [MESOS-1739](https://issues.apache.org/jira/browse/MESOS-1739) Allow slave reconfiguration on restart
  - [CORE-1244](https://jira.mesosphere.com/browse/CORE-1244) Allow agent domain re-configuration without losing tasks

## Checklist for all PRs

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by lava** [comment]
@mesosphere-mergebot bump-ee

**Comment by lava** [comment]
@vinodkone @adam-mesos  Could you review this?

**Comment by adam-mesos** [comment]
Q1. Do we actually want this enabled by default for all customers, even customers upgrading from 1.10 where they may have relied on the previous behavior?
Q2. Even if we want it enabled by default, should we add a config.yaml parameter so that customers can selectively disable this new behavior? Is there a chance somebody won't want it?

**Comment by vinodkone** [comment]
Making it configurable via config.yaml sounds reasonable to me. As far as
what should be the default, it should definitely be `additive` otherwise
upgrades to 1.11 EE won't work (since that sets the fault domains).

**Comment by lava** [comment]
@adam-mesos The old behaviour in this case would be for the agent to crash and do nothing until manual intervention occurs, so I doubt anyone depends on that.

@vinodkone I was looking through the existing integration tests but didn't really see anything that does a similar thing (I assume the test that we want is "Run a 1.10 cluster w/o fault domains and upgrade to an 1.11 cluster w/ fault domains configured.") Is there maybe an existing set of upgrade tests somewhere that would be a better fit for this?

**Comment by vinodkone** [comment]
@lava there is an upgrade test that @mnaboka is working on. so we probably don't need another one.

File: gen/dcos-config.yaml [hunk 1]
```diff
@@ -459,6 +459,7 @@ package:
       MESOS_LAUNCHER_DIR=/opt/mesosphere/active/mesos/libexec/mesos
       MESOS_EXECUTOR_ENVIRONMENT_VARIABLES=file:///opt/mesosphere/etc/mesos-executor-environment.json
       MESOS_EXECUTOR_REGISTRATION_TIMEOUT=10mins
+      MESOS_RECONFIGURATION_POLICY=additive
```
Comment (adam-mesos) [inline gen/dcos-config.yaml]: I can't find this flag in `docs/configuration/agent.md`, perhaps it only got added to `agent/flags.*`?

Comment (lava) [reply gen/dcos-config.yaml]: You're correct, the corresponding review isn't merged yet: https://reviews.apache.org/r/64450/

Naturally I'm hoping to get it committed as soon as possible, but with the current 1.11 frenzy going on I'm not sure how soon that will be.

**Comment by vinodkone** [comment]
Looks like @adam-mesos and myself approved this PR. Is there anything else the core team needs to do to get this merged? cc @orsenthil

**Comment by adam-mesos** [comment]
@vinodkone By the CI status, `mergebot/enterprise/review/approved/min_2` is still waiting for a second approval on the EE PR: https://github.com/mesosphere/dcos-enterprise/pull/1827

**Pull request merged**